autoload/go/jobcontrol.vim

```diff
@@ -93,7 +93,7 @@ endfunction
 " it'll be closed.
 function! s:on_exit(job_id, data)
   let std_combined = self.stderr + self.stdout
-  if empty(std_combined)
+  if jobwait([a:job_id])[0] > 0
```

Comment:
`jobwait(...)` is needed to get the return code of the job. Since it's called on_exit of the respective job, it should never block, so I didn't provide a timeout.